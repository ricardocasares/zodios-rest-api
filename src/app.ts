import { zodiosApp } from "@zodios/express";
import { spec } from "@app/api";
import { convert } from "@app/convert";

export const app = zodiosApp(spec).get("/temperature", (req, res) =>
  res.json({
    scale: req.query.to,
    value: convert(req.query.to, req.query.value),
  })
);
