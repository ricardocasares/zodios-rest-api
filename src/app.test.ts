import supertest from "supertest";
import { app } from "@app/app";

test("no parameters", () =>
  supertest(app)
    .get("/temperature")
    .expect(400)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("default parameter value: to", () =>
  supertest(app)
    .get("/temperature")
    .query({ value: 37 })
    .expect(200)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("requires parameter: value", () =>
  supertest(app)
    .get("/temperature")
    .query({ to: "F" })
    .expect(400)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("converts from C to F", () =>
  supertest(app)
    .get("/temperature")
    .query({ to: "C", value: 98.6 })
    .expect(200)
    .expect(({ body }) => expect(body).toMatchSnapshot()));

test("converts from F to C", () =>
  supertest(app)
    .get("/temperature")
    .query({ to: "C", value: 98.6 })
    .expect(200)
    .expect(({ body }) => expect(body).toMatchSnapshot()));
