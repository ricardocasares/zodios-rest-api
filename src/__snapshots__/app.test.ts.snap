// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts from C to F 1`] = `
{
  "scale": "C",
  "value": 37,
}
`;

exports[`converts from F to C 1`] = `
{
  "scale": "C",
  "value": 37,
}
`;

exports[`default parameter value: to 1`] = `
{
  "scale": "F",
  "value": 98.6,
}
`;

exports[`no parameters 1`] = `
{
  "context": "query.value",
  "error": [
    {
      "code": "invalid_type",
      "expected": "number",
      "message": "Required",
      "path": [],
      "received": "undefined",
    },
  ],
}
`;

exports[`requires parameter: value 1`] = `
{
  "context": "query.value",
  "error": [
    {
      "code": "invalid_type",
      "expected": "number",
      "message": "Required",
      "path": [],
      "received": "undefined",
    },
  ],
}
`;
